packages/purpose-template-process/src/services/purposeTemplateService.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  PurposeTemplate,
+  PurposeTemplateId,
+  WithMetadata,
+} from "pagopa-interop-models";
+import {
+  AppContext,
+  DB,
+  M2MAdminAuthData,
+  M2MAuthData,
+  UIAuthData,
+  WithLogger,
+} from "pagopa-interop-commons";
+import { ReadModelServiceSQL } from "./readModelServiceSQL.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function purposeTemplateServiceBuilder(
+  _dbInstance: DB,
+  readModelService: ReadModelServiceSQL
+) {
+  // TODO : use it to write purpose template events in the event store
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  // const repository = eventRepository(dbInstance, purposeEventToBinaryDataV2);
+
+  return {
+    async getPurposeTemplateById(
+      id: PurposeTemplateId,
+      {
+        logger,
+      }: WithLogger<AppContext<UIAuthData | M2MAuthData | M2MAdminAuthData>>
+    ): Promise<WithMetadata<PurposeTemplate>> {
+      logger.info(`Retrieving purpose template ${id}`);
+      return readModelService.getPurposeTemplateById(id);
```

Comment:
Follow the pattern from other services, where we don't throw in the readModelService and, instead, we have a method called `retrieveXX` that calls the readmodel and throws if the result is undefined (see also comments in the readModelService below)

```suggestion
      return retrievePurposeTemplate(id, readModelService);
```